microfast.F90

```diff
@@ -47,11 +47,13 @@ subroutine microfast(carma, cstate, iz, scale_threshold, rc)
   3 format(/,'microfast::ERROR - excessive change in supersaturation for ',a,' : iz=',i4,',lat=', &
               f7.2,',lon=',f7.2,',supsatiold=',e10.3,',supsatlold=',e10.3,',supsati=',e10.3, &
               ',supsatl=',e10.3,',t=',f6.2)
-  
+
    ! Set production and loss rates to zero.
   call zeromicro(carma, cstate, iz, rc)
   if (rc < RC_OK) return
-  
+
+  call coremasscheck( carma, cstate, iz, .false.,.true.,.true., "AfterZeromicro", rc )
```

Comment:
Why are there calls like this all the time in the code? If you are not going to fix, then this seems like it should only be on for debug. Should probably control this from the CARMA object.

Reply:
P.S. This occurs in many places, but I won't highlight them all.

Reply:
@zywshoon

Reply:
Suggest to put a control around this flag. Add a flag in the CARMA object -> could use the DEBUG flag.
@zywshoon

Reply:
debug flag added , done